### PR TITLE
Corrected the hinted return type for :func:`angle_between_vectors`

### PR DESCRIPTION
**Pull request by rayyan21d**
## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by AlexKempen** [comment]
This PR changes the return type of `angle_between_vectors` to `np.float64` (as opposed to `float`). Looking over the library, it seems like `float` is the default type hint for most floating point cases (including angles, like with `angle_of_vector`), whereas `np.float64` is not, as `np.float64` is hinted in only one other place in the library (the return type to `Mobject`'s `get_x`, `get_y`, and `get_z` functions). Thus, for consistency `float` probably makes more sense than `np.float64` here, even though `np.float64` is presumably more technically accurate.

I'll also point out (in case it wasn't obvious) that this PR is tied to this issue - someone with more permissions than me can link it if they wish: [https://github.com/ManimCommunity/manim/issues/3188](https://github.com/ManimCommunity/manim/issues/3188)

File: manim/utils/space_ops.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def angle_of_vector(vector: Sequence[float] | np.ndarray) -> float:
     return np.angle(complex(*vector[:2]))
 
 
-def angle_between_vectors(v1: np.ndarray, v2: np.ndarray) -> np.ndarray:
+def angle_between_vectors(v1: np.ndarray, v2: np.ndarray) -> np.float64:
```
Comment (behackl) [inline manim/utils/space_ops.py:327]:
```suggestion
def angle_between_vectors(v1: np.ndarray, v2: np.ndarray) -> float:
```

File: manim/utils/space_ops.py [hunk 2]
```diff
@@ -337,7 +337,7 @@ def angle_between_vectors(v1: np.ndarray, v2: np.ndarray) -> np.ndarray:
 
     Returns
     -------
-    np.ndarray
+    np.float64
```
Comment (behackl) [inline manim/utils/space_ops.py:340]:
```suggestion
    float
```

**Review by behackl — COMMENTED** [review]
+1 to @AlexKempen's comment, the type hints should be changed to `float`. I'll do so and merge this; thanks!

**Review by behackl — APPROVED** [review]

**Pull request merged**